### DISCUSSION TITLE
Making Sound Clips

### DISCUSSION THREAD

**Discussion by quietlight** [kitzeslab/opensoundscape#635] [Q&A] [answered]
Hi
Apologies for taking time of busy people. I am trying to make bulk sound clips but the for loop seems to conclude after 538 files instead of over a million. Python is not my strong point, I spend my time in Julia, but your work can help my project a lot. 

Find my static project pluto notebook here: [skraak.kiwi](https://skraak.kiwi), I have terrabytes of data from a small island.

Failed Jupyter notebook here:
[MakeClips1-onlydidfirst538clips.zip](https://github.com/kitzeslab/opensoundscape/files/10179590/MakeClips1-onlydidfirst538clips.zip)

This data set is pretty consistent, my files are labelled in AviaNZ which produces json label files, I culled 6091 15 minute files containing kiwi calls and rewrote the labels into raven files consistent with your example.

I need to train a binary classifier for haast tokoeka (kiwi), avianz gets around 20-40% true positives with a model trained for the Dept of Conservation, and the training interface is broken. I will later train a recogniser for my labels to save me work. M/F/MF/?, Near/Mid/Far, Noise, and possibly auto detect duets.

I add data at almost 1TB/month so have to get organised with a better recogniser.

I like the simple modular and extensible approach this project takes, also it does not lock me in, I can use my own tools written in Julia.

So the big problem for me here: why is that for loop dying?

Regards
David Cary

Ok, it is obvious when you look. Sorry for wasting time

**Comment by sammlapp** [comment]
I'm glad you found a resolution, don't hesitate to post questions about the code base or reach out with other questions in the future.

**Reply by quietlight** [reply, marked as answer]
Great thanks, I trained a very good recogniser first try. It gets a lot more calls than I was used to, but needs work with negative examples to weed out more false positives, its very good with frogs now, but distant frogs no, creaking trees, no, and the dawn chorus, no, I stop predicting at civil twilight and get lots of kiwi in the dawn chorus. 

I have something like 4000 calls to label, it will be some time. But eventually (in the next 2 months) I am going to train a male-kiwi/female-kiwi (haast tokoeka) recogniser so that I can mostly automate labelling. probably only duets will need hand annotation, after false positives have been weeded.

The simple naive julia function I use to segment the preds.csv into actual detections is extremely good at giving an accurate start and end, given that my predictions are made with 5s files with a 2.5s overlap. Its a million times better than what I was used to. 

Best thing though, pytorch blocks out my gpu and all the cores, I can process a 64gb sd card in 1hr, compared to 14 hours with avianz (I was running 3 at once, but still, no comparison).

I still use avianz for fine grained labelling because I like their json labels a lot, they have a json spec and I follow it. Most of my labelling workflow is in Finder using native mac tags on video files (munge png and wav with ffmpeg) because it is very fast and it even has keyboard shortcuts for tags.

I hope to get some more moths soon to expand into daylight bird monitoring, on 2 pest free or almost pest free islands. I was planning on 30 but its going to be less due to cost.

Thanks for a great project. If I can contribute anything I will, but python is not my first language.

Keep it up. Keep warm.

David
